core/modules/cart/helpers/syncCartWhenLocalStorageChange.ts

```diff
@@ -1,10 +1,18 @@
-import rootStore from '@vue-storefront/core/store';
+import rootStore from '@vue-storefront/core/store'
+import { storeViews } from 'config'
+import { currentStoreView } from '@vue-storefront/core/lib/multistore'
+
+function checkMultistoreKey (key: string, path: string): boolean {
+  const { multistore, commonCache } = storeViews
+  if (!multistore || (multistore && commonCache)) return key === path
+  return key === `${currentStoreView().storeCode}-${path}`
+}
 
 function getItemsFromStorage ({ key }) {
   const value = JSON.parse(localStorage[key])
```

Comment:
We still need to fetch the correct key in `JSON.parse(localStorage[key])` when multistore is enabled.

Reply:
I've moved it into the if-else clause.